eventsourcing/src/main/java/org/axonframework/eventsourcing/AnnotationBasedEventStateApplier.java

```diff
@@ -0,0 +1,117 @@
+/*
+ * Copyright (c) 2010-2025. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.eventsourcing;
+
+import org.axonframework.eventhandling.EventMessage;
+import org.axonframework.messaging.MessageStream;
+import org.axonframework.messaging.annotation.AnnotatedHandlerInspector;
+import org.axonframework.messaging.annotation.ClasspathHandlerDefinition;
+import org.axonframework.messaging.annotation.ClasspathParameterResolverFactory;
+import org.axonframework.messaging.unitofwork.ProcessingContext;
+
+import java.util.concurrent.CompletableFuture;
+import javax.annotation.Nonnull;
+
+import static java.util.Objects.requireNonNull;
+
+
+/**
+ * Implementation of {@link EventStateApplier} that applies state changes through {@link EventSourcingHandler} annotated
+ * methods using an {@link AnnotatedHandlerInspector}.
+ *
+ * @param <M> The type of model to apply state changes to
+ * @author Mateusz Nowak
+ * @see EventSourcingHandler
+ * @see AnnotatedHandlerInspector
+ * @since 5.0.0
+ */
+public class AnnotationBasedEventStateApplier<M> implements EventStateApplier<M> {
+
+    private final AnnotatedHandlerInspector<Object> inspector;
```

Comment:
```suggestion
    private final AnnotatedHandlerInspector<M> inspector;
```
The type can be in here right?